src/Microsoft.AspNetCore.Razor.Language/Legacy/MarkupParser.cs

```diff
@@ -0,0 +1,916 @@
+                }
+                TryAccept(SyntaxKind.Text);
+
+                // Parse the contents of a tag like attributes.
+                ParseTagContent(tagBuilder);
+
+                if (TryAccept(SyntaxKind.ForwardSlash))
+                {
+                    // This is a self closing tag.
+                    expectsBody = false;
+                }
+                TryAccept(SyntaxKind.CloseAngle);
+
+                // End tag block
+                tagBuilder.Add(OutputAsMarkupLiteral());
+                var tagBlock = SyntaxFactory.MarkupTagBlock(tagBuilder.ToList());
+                return tagBlock;
+            }
+        }
+
+        private MarkupTagBlockSyntax ParseEndTag()
+        {
+            // This section can accept things like: '</p  >' or '</p>' etc.
+            Assert(SyntaxKind.OpenAngle);
+
+            using (var pooledResult = Pool.Allocate<RazorSyntaxNode>())
+            {
+                var tagBuilder = pooledResult.Builder;
+
+                AcceptAndMoveNext(); // Accept '<'
+                TryAccept(SyntaxKind.ForwardSlash);
+
+                // Whitespace here is invalid (according to the spec)
+                ParseOptionalBangEscape(tagBuilder);
+                TryAccept(SyntaxKind.Text);
+                TryAccept(SyntaxKind.Whitespace);
+                TryAccept(SyntaxKind.CloseAngle);
+
+                // End tag block
+                tagBuilder.Add(OutputAsMarkupLiteral());
+                var tagBlock = SyntaxFactory.MarkupTagBlock(tagBuilder.ToList());
+                return tagBlock;
+            }
+        }
+
+        private SyntaxList<RazorSyntaxNode> ParseTagBody(string tagName, out bool seenEndTag)
+        {
+            // No-op here for now.
+            seenEndTag = false;
+            return null;
+        }
+
+        private string GetTagName(MarkupTagBlockSyntax tagBlock)
+        {
+            var child = tagBlock.Children[0];
+
+            if (tagBlock.Children.Count == 0 || !(child is MarkupTextLiteralSyntax))
+            {
+                return null;
+            }
+
+            var childLiteral = (MarkupTextLiteralSyntax)child;
+            SyntaxToken textToken = null;
+            for (var i = 0; i < childLiteral.LiteralTokens.Count; i++)
+            {
+                var token = childLiteral.LiteralTokens[i];
+
+                if (token != null &&
+                    (token.Kind == SyntaxKind.Whitespace || token.Kind == SyntaxKind.Text))
+                {
+                    textToken = token;
+                    break;
+                }
+            }
+
+            if (textToken == null)
+            {
+                return null;
+            }
+
+            return textToken.Kind == SyntaxKind.Whitespace ? null : textToken.Content;
+        }
+
+        private void ParseTagContent(in SyntaxListBuilder<RazorSyntaxNode> builder)
+        {
+            // Calling to legacy code for now.
+            LegacyParseTagContent(builder);
```

Comment:
In my next PR this will go away and I'll refactor attribute parsing and move it here.